src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
Can we add this to the stable version of 3.1? I just updated to 3.1 yesterday via composer, and noticed this error in my app. I came here and found this fix. I added it to my AppController (and overloaded the render method) for now, since I don't want to modify the cake code.